### ISSUE TITLE
Inconsistent tab focus when selecting concept from sidebar

### ISSUE THREAD

**Issue by osma** [NatLibFi/Skosmos#1086]
## At which URL did you encounter the problem?

http://finto.fi/yso/en/page/p38965

## What steps will reproduce the problem?
1. Select a concept from the A-Z (alphabetical) or New tab
2. Select the same concept from the Hierarchy or Groups tab

## What is the expected output? What do you see instead?

Expected that tab focus would work the same way regardless of which sidebar tab is used to select a concept.

In the A-Z and New tabs, when a concept is selected, tab focus is left on the link, which is shown with a dotted outline:

![image](https://user-images.githubusercontent.com/1132830/97178260-eab68080-179f-11eb-9d12-74de5f41f8e3.png)

In the Hierarchy and Group tabs, when selecting a concept, the tab focus instead moves to the concept information display, so no dotted outline is shown for the concept link.

The behaviour should be consistent: either the tab focus should move to the concept display in all cases (meaning that no dotted outline is shown), or it should always remain on the concept link, meaning that the dotted outline would also be shown in the Hierarchy and Groups tabs.

This is partly a consequence of the accessibility improvements done in release 2.8. Previously tab focus was not shown at all, so although the behaviour was different, it wasn't obvious just looking at the UI in a normal web browser.

## What browser did you use? (eg. Firefox, Chrome, Safari, Internet explorer)

Firefox

**Comment by kouralex** [comment]
From the perspective of accessibility, (tab) focus should not (in my opinion) change unexpectedly, i.e., after an AJAX request has been completed after some time.

There are couple of partially related problems here and around the aforementioned elements, some of which should most likely be their own issues:

1) The jstree library that is responsible for showing the Hierarchy and Groups tab does not, at the moment, properly support normal keyboard usage (e.g., tabbing).

2) Selecting (clicking) a letter in alphabetical index or clicking any of the nav-tabs (A-Z, Hierarchical, Groups, New) in sidebar : response replaces the data in the complete sidebar block where the tab focus may or may not be at the very same moment - resulting in lost focus in the latter case.

3) Clicking Hierarchy tab works a bit differently, it forces in many cases a full page load for no apparent reason.

4) The custom scrollbar does not follow focus changes as one would expect in A-Z and New tabs - only mouse scrolling works as intended (e.g., scrollbar disappears and it is possible to have focus out of view).

5) Clicking on "empty space" does not set the tab focus (invisibly) there, e.g., in the alphabetical index (pressing tab afterwards causes the focus to be in very first list element instead of the closest element with respect to the clicked area; IMHO this is the default browser action, at least in the case of intermingled HTML text and link elements [NB: this may not be the case here, though]). The very same applies to .concept-info and even widgets in case a focus capturing element is not clicked. It seems like setting the `tabindex` attribute on an ancestor element (e.g., set on #maincontent and .mCustomScrollBox) affects this, at least.

That said, I believe fixing #1086 will be easier with changing the tab focus to content (concept data) on AJAX completion but that is not necessarily the maximally accessible approach.

**Comment by miguelahonen** [comment]
The descriptions of the web accessibility guidelines (WCAG21) on the page https://www.w3.org/WAI/WCAG21/Understanding/consistent-navigation.html mention the criterion related to consistency 

Understanding Success Criterion 3.2.3: Consistent Navigation:
_The intent of this Success Criterion is to encourage the use of consistent presentation and layout for users who interact with repeated content within a set of webpages and need to locate a specific information or functionality more than once..._
